README.md

```diff
@@ -8,27 +8,194 @@
+Setting up the full OAuth2 authorization code grant workflow is beyond
+the scope of this lesson, but it is described well in the Github
+[docs](https://developer.github.com/v3/oauth/).
 
-If there's time (after the intro, getting them into the promise-like syntax, etc.), it might be worth having them sign up for a service and pass tokens in the `headers` part of the options, e.g.,
+Fortunately for us, Github also allows you to generate your own personal
+authorization token that we can use to give us authorized access to the
+API. And `fetch` makes it super easy to implement.
 
-```javascript
-fetch(`https://api.github.com/users/${username}`, { 
-  headers: { 
-    Authorization: 'token ${token}'
+To start, go to https://github.com/settings/tokens and click "Generate
+new token." Name it "Learn.co" and check `repo` scope. Once you generate
+the token, make sure to copy and paste it somewhere, because once you
+leave that page, you won't be able to see it again. This is for your
+security - even if someone were to gain access to this page on your
+account, they still couldn't see your tokens.
+
+Using the token to [access the API](https://developer.github.com/v3/oauth/#3-use-the-access-token-to-access-the-api) is a simple matter of creating an `Authorization` header with our request. Let's try it out by listing our repos.
+
+```js
+fetch("https://api.github.com/user/repos").
+  then(res => res.json()).
+  then(json => console.log(json))
+```
+
+If we run that code, we'll get a `401 (Unauthorized)` response.
+We need to provide our authorization token in order to list our own
+repositories with this API, so let's add our `Authorization` header
+(don't forget to assign your token to `const token`).
+
+```js
+const token = "YOUR_TOKEN_HERE"
+fetch(`https://api.github.com/user/repos`, {
+  headers: {
+    Authorization: `token ${token}`
   }
 }).then(res => res.json()).then(json => console.log(json))
 ```
 
-As they'll be doing this with Twitter in the next lesson anyway
+We just pass the desired headers as part of a second options argument to
+`fetch` and we are in business. Easy as that!
+
+![so fetch](http://i.giphy.com/SUgOYsXqmexxe.gif)
+
+**Top-Tip:** Don't ever give out your access token or store it in a
+publicly accessible place or a shared Github repository. We're just using these for learning purposes. In a production setting, user's access tokens would be stored securely in a database and not exposed to other people.
+
+## Caveat Browser
+
+As you can see, `fetch` provides us with such a clean, low-maintenance
+way to fetch and work with resources. You may be wondering why you'd
+ever use XHR again.
+
+Keep in mind that, while it is increasing, [browser support](http://caniuse.com/#feat=fetch) for `fetch` is still limited primarily to current versions of Chrome, Firefox, and Opera. So if you're supporting older browsers, don't let XHR and jQuery Ajax go just yet. They're still powerful and useful tools for creating dynamic applications.
```

Comment:
Good to note. I'm tempted to point students to https://github.com/github/fetch, but I think you made a good call here -- the polyfill is out of scope.